### PR TITLE
testing, formatting and updated examples

### PR DESCRIPTION
**Pull request by mhsjlw**
needs to be tested further

### CODE REVIEW

**Comment by rom1504** [comment]
I'll test it but it looks good to me.

**Comment by rom1504** [comment]
@mhsjlw the branch has conflict, can't merge, can you rebase ?

File: src/createClient.js [hunk 1]
```diff
@@ -9,26 +9,28 @@ module.exports = createClient;
...
-    client.emit("connect");
+
+  socket.on('listening', function() {
+    client.emit('connected');
   });
 
   client.setSocket(socket);
 
-  client.on("connect",onConnect);
+  client.on('comnected', onConnect);
```
Comment (rom1504) [inline src/createClient.js]: comnected

File: src/client.js [hunk 2]
```diff
@@ -38,41 +35,43 @@ class Client extends EventEmitter
...
+      }
+
+      e.field = parts.join('.');
+      e.message = `Serialization error for ${e.field}: ${e.message}`;
+      this.emit('error', e);
     });
 
 
-    this.parser.on('error', (e) => {
+    this.parser.on('error', function(e) {
```
Comment (rom1504) [inline src/client.js]: why do you remove arrow functions ?
There's no reason to and it will break the `this`

**Comment by rom1504** [comment]
would be useful to rebase this @mhsjlw

**Comment by mhsjlw** [comment]
yup, I'll do it when I get a chance and set up circle ci too

**Comment by mhsjlw** [comment]
It seems that this PR is really broken, I'll just open a new one.

**Pull request closed without merging**